### PR TITLE
paddle::framework::vectorize() templatization [PART2]

### PR DESCRIPTION
**Pull request by luotao1**
Suggested in https://github.com/PaddlePaddle/Paddle/pull/19510 changes to `paddle::framework::vectorize()` function. Following changes has allowed to get rid of `vectorize2int() `function from mkldnn ops.

TODO： remove `std::vector<int> vectorize2int(const DDim& ddim)` in next PR due to file limits.

### CODE REVIEW

**Comment by luotao1** [comment]
@grygielski @jczaja Please take a review!

File: paddle/fluid/operators/fused/fusion_conv_inception_op.cu [hunk 1]
```diff
@@ -61,7 +61,7 @@ class CUDNNConvInceptionFusionOpKernel : public framework::OpKernel<T> {
     T* temp_data = temp_outs[0]->mutable_data<T>(input->dims(), ctx.GetPlace());
 
     DataLayout layout = DataLayout::kNCHW;
-    std::vector<int> in_dim = framework::vectorize2int(input->dims());
+    std::vector<int> in_dim = framework::vectorize<int>(input->dims());
```
Comment (grygielski) [inline paddle/fluid/operators/fused/fusion_conv_inception_op.cu:64]: I'm not sure if You want to dive so much into changes but some lines might be improved by using `auto` keyword

**Review by grygielski — DISMISSED** [review]
Please look at my comment, but: LGTM

File: paddle/fluid/operators/fused/fusion_conv_inception_op.cu (hunk 1 above)
Comment (luotao1) [reply paddle/fluid/operators/fused/fusion_conv_inception_op.cu:64]: I use one `sed` command to do this change, so I don't use `auto` keyword.

**Review by sneaxiy — APPROVED** [review]
LGTM.

**Pull request merged**